cookbooks/aws-parallelcluster-slurm/resources/munge/partial/_munge_actions.rb

```diff
@@ -0,0 +1,118 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Munge
+munge_version = '0.5.14'
+munge_url = "https://github.com/dun/munge/archive/munge-#{munge_version}.tar.gz"
+munge_tarball = "#{node['cluster']['sources_dir']}/munge-#{munge_version}.tar.gz"
+munge_user = 'munge'
+munge_user_id = node['cluster']['reserved_base_uid'] + 2
+munge_group = munge_user
+munge_group_id = munge_user_id
+
+action :setup do
+  actions = lambda {
+    action_purge_packages
+    action_download_source_code
+    action_compile_and_install
+    action_update_init_script
+    action_set_user_and_group
+    action_create_required_directories
+  }
+  actions.call unless redhat_ubi?
+end
+
+action :purge_packages do
+  package %w(munge* libmunge*) do
+    action :purge
+  end
+end
+
+action :download_source_code do
+  # Get munge tarball
+  remote_file munge_tarball do
+    source munge_url
+    mode '0644'
+    retries 3
+    retry_delay 5
+    # TODO: Add version or checksum checks
+    action :create_if_missing
+  end
+end
+
+action :compile_and_install do
+  # Install munge
+  bash 'make install' do
+    user 'root'
+    group 'root'
+    cwd Chef::Config[:file_cache_path]
+    code <<-MUNGE
+      set -e
+      tar xf #{munge_tarball}
+      cd munge-munge-#{munge_version}
+      ./bootstrap
+      ./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var --libdir=#{munge_libdir}
+      CORES=$(grep processor /proc/cpuinfo | wc -l)
+      make -j $CORES
+      make install
+    MUNGE
+    not_if "/usr/sbin/munged --version | grep -q munge-#{munge_version} && ls #{munge_libdir}/libmunge*"
+  end
+end
+
+action :update_init_script do
+  # Updated munge init script for Amazon Linux
+  template '/etc/init.d/munge' do
+    source 'munge/munge-init.erb'
+    owner 'root'
+    group 'root'
+    variables(munge_user: munge_user,
+              munge_group: munge_group)
+    mode '0755'
+  end
+end
+
+action :set_user_and_group do
+  # Setup munge group
+  group munge_group do
+    comment 'munge group'
+    gid munge_group_id
+    system true
+  end
+
+  # Setup munge user
+  user munge_user do
+    uid munge_user_id
+    gid munge_group_id
+    manage_home false
+    comment 'munge user'
+    system true
+    shell '/sbin/nologin'
+  end
+end
+
+action :create_required_directories do
+  # Create required directories for munge
+  dirs = ["/var/log/munge", "/etc/munge", "/var/run/munge"]
+  dirs.each do |dir|
+    directory dir do
+      action :create
+      owner munge_user
+      group munge_group
+    end
+  end
+end
```

Comment:
I believe here we are missing the [setup_munge_head_node](https://github.com/aws/aws-parallelcluster-cookbook/blob/1a57f16b59968528609ae51765ce98e4405f7d76/libraries/helpers.rb#L409-L422), which I believe functionally belongs to this `munge` resource. Should we move it here and then use it in `config_head_node` as a `generate_munge_key` action of the `munge` resource?

Reply:
Actually there are additional operations performed in that helpers.rb.

We can even leave them where they are and move them to the new resource at a later stage if we prefer (but this refactoring seems the best opportunity to me).

Reply:
These are part of another task and will be addressed in another PR.